src/postamble.js

```diff
@@ -161,6 +161,14 @@ function callMain(args) {
     abortWrapperDepth += 1;
 #endif
 
+#if ASYNCIFY == 2
+    // Keep the runtime alive until the entry function promise has resolved.
+    // In ASYNCIFY == 1 mode, this is not needed because it calls an
+    // extra runtimeKeepalivePush when suspending to prevent the maybeExit call
+    // from exiting after resuming.
```

Comment:
In what function does `ASYNCIFY==1` do that? Perhaps we can do the `ASYNCIFY==2` 
workaround there as well, I'm hoping.

(Does this match the pop down below?)

Reply:
The extra push comes from `maybeStopUnwind` and pop comes from `doRewind` with `ASYNCIFY==1`. None of the rewind/unwind code is used in the new mode. We could push/pop in an extra wrapper function, but since we now have a promise that represents when something is done I don't really see why we'd want this. 

In the new JSPI mode we could probably get rid of the runtimekeepalive stuff altogether and rely on the promise, but I'd like to do that when we separate out this file for each mode.

Reply:
I see, thanks. sgtm.